pg_shard.c

```diff
@@ -579,7 +579,7 @@ ExtractFirstDistributedTableId(Query *query)
 	{
 		RangeTblEntry *rangeTableEntry = (RangeTblEntry *) lfirst(rangeTableCell);
 
-		if (IsDistributedTable(rangeTableEntry->relid))
+		if (rangeTableEntry->rtekind == RTE_RELATION && IsDistributedTable(rangeTableEntry->relid))
```

Comment:
What is the impact of passing relid to 0 to IsDistributedTable? Does it crash? I'd think it would be OK to pass 0 as that should just return false?

- Sumedh